cmd/kubermatic-installer/cmd_mirror_binaries.go

```diff
@@ -0,0 +1,423 @@
+}
+
+func verifyChecksum(ctx context.Context, checksumUrl string, binaryFilePath string) error {
+	expectedChecksum, err := getChecksumFromURL(ctx, checksumUrl)
+	if err != nil {
+		return fmt.Errorf("error getting checksum from url %s: %w", checksumUrl, err)
+	}
+	actualChecksum, err := getChecksumOfFile(binaryFilePath)
+	if err != nil {
+		return fmt.Errorf("error getting checksum of file %s: %w", binaryFilePath, err)
+	}
+	if expectedChecksum != actualChecksum {
+		return fmt.Errorf("checksum verification failed for %s: expected %s, got %s", binaryFilePath, expectedChecksum, actualChecksum)
+	}
+	return nil
+}
+
+func getCriToolsRelease(version semverlib.Version) (string, error) {
+	release := fmt.Sprintf("%d.%d", version.Major(), version.Minor())
+	var criToolsReleases = map[string]string{
+		"1.32": "v1.32.0",
+		"1.31": "v1.31.1",
+		"1.30": "v1.30.1",
+		"1.29": "v1.29.0",
+		"1.28": "v1.28.0",
+		"1.27": "v1.27.1",
+		"1.26": "v1.26.1",
+		"1.25": "v1.25.0",
+		"1.24": "v1.24.2",
+	}
+	if criToolRelease, ok := criToolsReleases[release]; ok {
+		return criToolRelease, nil
+	}
+
+	return "v1.32.0", nil
+}
+
+// downloadCriTools downloads the CRI tools tarball and its checksum for the given Kubernetes version.
+func downloadCriTools(ctx context.Context, logger *logrus.Logger, version semverlib.Version, binPath, hostArch string) error {
+	criToolsRelease, err := getCriToolsRelease(version)
+	if err != nil {
+		return fmt.Errorf("failed to get CRI tools release for Kubernetes version %s: %w", version, err)
+	}
+
+	criToolsDir := filepath.Join(binPath, "kubernetes-sigs", "cri-tools", "releases", "download", criToolsRelease)
+
+	// Ensure the directory exists and is empty
+	if err := ensureCleanDir(criToolsDir); err != nil {
+		return fmt.Errorf("failed to prepare CRI tools directory: %w", err)
+	}
+
+	logger.Debugf("⏳ Downloading CRI tools %s...", criToolsRelease)
+
+	criToolsFileName := fmt.Sprintf("crictl-%s-linux-%s.tar.gz", criToolsRelease, hostArch)
+	criToolsURL := fmt.Sprintf("%s/%s/%s", CRIToolsBaseURL, criToolsRelease, criToolsFileName)
+	criToolsFilePath := filepath.Join(criToolsDir, criToolsFileName)
+
+	// Download tarball
+	if err := downloadFromUrl(ctx, criToolsURL, criToolsFilePath); err != nil {
+		return fmt.Errorf("failed to download CRI tools tarball (%s): %w", criToolsRelease, err)
+	}
+
+	// Download and save checksum file
+	checksumFileName := criToolsFileName + ".sha256"
+	checksumURL := fmt.Sprintf("%s/%s/%s", CRIToolsBaseURL, criToolsRelease, checksumFileName)
+	checksumFilePath := filepath.Join(criToolsDir, checksumFileName)
+
+	if err := downloadFromUrl(ctx, checksumURL, checksumFilePath); err != nil {
+		return fmt.Errorf("failed to download CRI tools checksum file (%s): %w", criToolsRelease, err)
+	}
+
+	// Verify checksum
+	if err := verifyChecksum(ctx, checksumURL, criToolsFilePath); err != nil {
+		return fmt.Errorf("failed to verify CRI tools tarball checksum (%s): %w", criToolsRelease, err)
+	}
```

Comment:
Since this is repeated multiple times, we can create a function downloadAndVerifyChecksum for this.